engine/src/main/scala/cromwell/engine/workflow/workflowstore/WorkflowStoreEngineActor.scala

```diff
@@ -77,11 +76,19 @@ final case class WorkflowStoreEngineActor(store: WorkflowStore, serviceRegistryA
       case cmd @ AbortWorkflow(id, manager) =>
         store.remove(id) map { removed =>
           if (removed) {
-            log.debug(s"Workflow $id aborted and removed from the workflow store.")
             manager ! WorkflowManagerActor.AbortWorkflowCommand(id, sndr)
+            log.debug(s"Workflow $id removed from the workflow store, abort requested.")
           } else {
             sndr ! WorkflowAbortFailed(id, new WorkflowNotFoundException(s"Couldn't abort $id because no workflow with that ID is in progress"))
           }
+        } recover {
+          case t =>
+            val message = s"Error aborting workflow $id: could not remove from workflow store"
+            log.error(t, message)
+            // A generic exception type like RuntimeException will produce a 500 at the API layer, which seems appropriate
+            // given we don't know much about what went wrong here.  `t.getMessage` so the cause propagates to the client.
+            val e = new RuntimeException(s"$message: ${t.getMessage}", t)
+            sndr ! WorkflowAbortFailed(id, e)
```

Comment:
This is different from the submission fix.  In this case there was no API response, so the client may have assumed that the abort was successful.  We now make it clear that it was not, so hopefully the client will try again with better results.